play/src/front/Phaser/Game/GameScene.ts

```diff
@@ -2135,146 +2126,156 @@ export class GameScene extends DirtyScene {
             //this.reposition();
         });
 
+        if (!this.peerStoreUnsubscriber) {
+            this.peerStoreUnsubscriber = [];
+        }
+
         const talkIconVolumeTreshold = 10;
         let oldPeersNumber = 0;
         let oldUsers = new Map<number, MessageUserJoined>();
         let screenWakeRelease: (() => Promise<void>) | undefined;
-
         let alreadyInBubble = false;
         const pendingConnects = new Set<number>();
-        this.peerStoreUnsubscriber = peerStore.subscribe((peers) => {
-            const newPeerNumber = peers.size;
-            const newUsers = new Map<number, MessageUserJoined>();
-            const players = this.remotePlayersRepository.getPlayers();
-
-            for (const playerId of peers.keys()) {
-                const currentPlayer = players.get(playerId);
-                if (currentPlayer) {
-                    newUsers.set(playerId, currentPlayer);
-                }
-            }
-
-            // Join
-            if (oldPeersNumber === 0 && newPeerNumber > oldPeersNumber) {
-                // Note: by design, the peerStore can only add or remove one user at a given time.
-                // So we know for sure that there is only one new user.
-                const peer = Array.from(peers.values())[0];
-                //askIfUserWantToJoinBubbleOf(peer.userName);
-                statusChanger.setUserNameInteraction(peer.player.name);
-                statusChanger.applyInteractionRules();
-
-                pendingConnects.add(peer.userId);
-                setTimeout(() => {
-                    // In case the peer never connects, we should remove it from the pendingConnects after a timeout
-                    pendingConnects.delete(peer.userId);
-                    /*if (pendingConnects.size === 0 && !alreadyInBubble && !this.cleanupDone) {
-                        iframeListener.sendJoinProximityMeetingEvent(Array.from(newUsers.values()));
-                        alreadyInBubble = true;
-                    }*/
-                }, 5000);
-                peer.once("connect", () => {
-                    pendingConnects.delete(peer.userId);
-                    if (pendingConnects.size === 0) {
-                        iframeListener.sendJoinProximityMeetingEvent(Array.from(newUsers.values()));
-                        alreadyInBubble = true;
-                    }
-                });
-            }
 
-            // Left
-            if (newPeerNumber === 0 && newPeerNumber < oldPeersNumber) {
-                // TODO: leave event can be triggered without a join if connect fails
-                hideBubbleConfirmationModal();
-                iframeListener.sendLeaveProximityMeetingEvent();
+        // Store all subscriptions to be able to unsubscribe later
+        this.peerStoreUnsubscriber.push(
```

Comment:
Get rid of the peerStore.

The ExtendedSpaceUser object could have a getVideoStreamStore / getScreenSharingStreamStore attached that returns a "Streamable" (i.e. for now a VideoPeer or ScreenSharingPeer and tomorrow a Livekit video)